portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,10 +38,17 @@
 public class DataServlet extends HttpServlet {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String currComment = request.getParameter("comment");
-    writeComments(currComment);
+    String originalText = request.getParameter("text");
+    String languageCode = request.getParameter("languageCode");
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
+    Translation translation =
+        translate.translate(originalText, Translate.TranslateOption.targetLanguage(languageCode));
+    String translatedText = translation.getTranslatedText();
+    writeComments(translatedText);
```

Comment:
You can move the translation code to a separate function

String translateComment(request) {
}

You can also add additional sanity checks,
1) check that the language is specified
2) Don't translate if the input comment text is empty. 
3) don't translate if the target language is same as the source language.

Reply:
done